worker/src/RTC/Transport.cpp

```diff
@@ -666,6 +666,34 @@ namespace RTC
 				break;
 			}
 
+			case RTC::RTCP::Type::XR:
+			{
+				auto* ser = static_cast<RTC::RTCP::SenderExtendedReportPacket*>(packet);
```

Comment:
You are assuming here that all incoming RTCP XR packets are composed merely by bocks of type 5. The packet could contain any kind of block instead, and this code will interpret all of them as if they where DLRR report blocks.

More on this later in the discussion section.